feat: add new script for 101-02 lab to query via JSON-RPC

bguiz opened the pull request.

bguiz committed:
feat: add new script for 101-02 lab to query via JSON-RPC

Signed-off-by: Brendan Graetz <bguiz@users.noreply.github.com>

bguiz committed:
docs: add README for lab, also template README for other labs within smoke test

Signed-off-by: Brendan Graetz <bguiz@users.noreply.github.com>

bguiz committed:
docs: add contents directory for module 101-02

Signed-off-by: Brendan Graetz <bguiz@users.noreply.github.com>

bguiz committed:
docs: create templates for the module materials

Signed-off-by: Brendan Graetz <bguiz@users.noreply.github.com>

bguiz committed:
docs: add empty template for module 101-02 student guide

Signed-off-by: Brendan Graetz <bguiz@users.noreply.github.com>

bguiz committed:
docs: write overview and learning outcomes for module 101-02

Signed-off-by: Brendan Graetz <bguiz@users.noreply.github.com>

bguiz committed:
docs: add code snippets and resources for module 101-02 student guide

Signed-off-by: Brendan Graetz <bguiz@users.noreply.github.com>

bguiz committed:
docs: add self assessment questions and answers to module 01-02 student guide

Signed-off-by: Brendan Graetz <bguiz@users.noreply.github.com>

bguiz committed:
docs: add template file for MCQs

Signed-off-by: Brendan Graetz <bguiz@users.noreply.github.com>

bguiz committed:
docs: added MCQs for module 101-02

Signed-off-by: Brendan Graetz <bguiz@users.noreply.github.com>

bguiz committed:
docs: add definitions and a few new resources in the module 101-02 student guide

Signed-off-by: Brendan Graetz <bguiz@users.noreply.github.com>

bguiz committed:
docs: add graded assessment for module 101-02

Signed-off-by: Brendan Graetz <bguiz@users.noreply.github.com>

bguiz committed:
docs: add template for graded assessments

Signed-off-by: Brendan Graetz <bguiz@users.noreply.github.com>

michielmulders commented on hdc10102LabQueryRpc/README.md:
```diff
@@ -0,0 +1,57 @@
+
+## Steps
+
+- In the terminal, start in the **root directory** of this repo
```
Instead of doing all the nested bullet points, can we use h3 headings and group steps? 
We can do 
h3] Step/Task 1: Retrieve block number
[write its contents...]

h3]  Step/Taks 2: ...

This feels less overwhelming than this wall of bullet points

michielmulders reviewed (changes requested):
One change requested regarding steps and task format

bguiz committed:
docs: clean up graded assessment instructions

- includes PR review feedback from @michielmulders - https://github.com/hedera-dev/hedera-dev-course-101/pull/2#discussion_r1878142803

Signed-off-by: Brendan Graetz <bguiz@users.noreply.github.com>

bguiz committed:
docs: make it easier to read the lab instructions for module 101-02

- adresses feedback from @michielmulders - https://github.com/hedera-dev/hedera-dev-course-101/pull/1#discussion_r1878129707

Signed-off-by: Brendan Graetz <bguiz@users.noreply.github.com>

michielmulders committed:
Merge pull request #2 from hedera-dev/feat/module-101-02-content

feat: module 101-02 content

michielmulders reviewed (approved):
Thanks!
